### PR TITLE
Make database version configurable

### PR DESCRIPTION
**Pull request by ajmaddaford**
### What is the context of this PR?
Makes the database version configurable. The default has been set to 9.4.9 as we use 9.4.x in production. 

### How to review
Run database Terraform and verify database version changes are applied in AWS

### CODE REVIEW

**Comment by jonnyshaw89** [comment]
Is it wise to switch between versions between environments. will 9.5.x run on t2 instances?

File: survey-runner-database/terraform.tfvars.example [hunk 1]
```diff
@@ -5,5 +5,6 @@ database_cidrs=["10.30.20.96/28","10.30.20.112/28"]
 
 multi_az = false
 backup_retention_period = 0
+database_allocated_storage = 10
```
Comment (jonnyshaw89) [inline survey-runner-database/terraform.tfvars.example:8]: this needs to be at least 100

**Review by jonnyshaw89 — CHANGES_REQUESTED** [review]

File: survey-runner-database/terraform.tfvars.example (hunk 1 above)
Comment (ajmaddaford) [reply survey-runner-database/terraform.tfvars.example:8]: No it doesn't 😛

Comment (jonnyshaw89) [reply survey-runner-database/terraform.tfvars.example:8]: gp2 minimum storage is 100 so i guess it defaults higher but its good to be explicit :)

Comment (ajmaddaford) [reply survey-runner-database/terraform.tfvars.example:8]: From http://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/CHAP_Storage.html#Concepts.Storage
> A gp2 volume can range in size from 1 GiB to 16 TiB.

File: survey-runner-database/global_vars.tf [hunk 2]
```diff
@@ -39,6 +39,16 @@ variable "database_allocated_storage" {
...
 }
 
+variable "database_engine_version" {
+  description = "The Postgres database engine version"
+  default     = "9.4.9"
+}
+
+variable "allow_major_version_upgrade" {
+    description = "Allow major database version upgrades e.g. 9.4.x to 9.5.x"
+    default     = true
```
Comment (benjefferies) [inline survey-runner-database/global_vars.tf]: Should this be false until we want to upgrade?

**Review by benjefferies — CHANGES_REQUESTED** [review]

File: survey-runner-database/global_vars.tf (hunk 2 above)
Comment (ajmaddaford) [reply survey-runner-database/global_vars.tf]: Done. I also changed the `database_apply_immediately` default to `false`.

**Review by benjefferies — APPROVED** [review]

File: survey-runner-database/terraform.tfvars.example (hunk 1 above)
Comment (jonnyshaw89) [reply survey-runner-database/terraform.tfvars.example:8]: I stand corrected. This must have changed from when they were first launched.

**Review by jonnyshaw89 — APPROVED** [review]

**Comment by jonnyshaw89** [comment]
@ajmaddaford rebase & squash?

**Pull request merged**